docs/markdown/DevGuide_BuildProcessor.md

```diff
@@ -0,0 +1,115 @@
+You can also use all your defined properties here. This let's you access all your algorithm's parameters directly from the GUI with automatic updates upon change.
+
+### Frequently used operations
+
+#### Setting up an `ImageOutport`:
+As mentioned above, most processors that output an image should also have an optional image inport for compositing of different renders. Usually by default you want to render on top of this incoming image, which basically holds an OpenGL framebuffer. That means you usually want to copy the inport data to the outport and activate it as target framebuffer for your rendering. `utilgl::activateTargetAndClearOrCopySource(outport, image_inport)` does exactly this for you. All your rendering will be automatically composited according to the framebuffer depth layer. Note that it is good practice to deactivate the target at the end of your `process()` by using `utilgl::deactivateCurrentTarget()`. You can include the function from `modules/opengl/texture/textureutils.h`.
+
+#### Shader and Texture utilities:
+The `utilgl` namespace includes more useful functions for OpenGL, like controlling various OpenGL states (see `modules/opengl/openglutils.h`), utilities for working with `Shader`s (see `modules/opengl/shader/shaderutils/h`), drawing full screen quads and more.
+
+The `Shader` class can be used to activate shaders and bind uniforms etc. Similarly you can use the `TextureUnit` class as a wrapper for texture units. An image inport can be bound to a texture unit using
+```
+TextureUnit tex_unit;
+utilgl::bindColorTexture(tex_inport, tex_unit.getEnum());
+shader.setUniform("Uniform Name", tex_unit.getUnitNumber()); // shader is initialized in constructor
+```
+
+#### Dispatching data to the CPU:
+Inviwo has different data representations, depending on where the data is stored (Hard drive, RAM, GPU RAM) and by which API it is used (e.g. OpenGL, OpenCL). You can request each of those representations using `getRepresentation<RepresentationType>()`. For example you could get a volume representation in RAM as follows:
+```
+auto volume_ram_repr = volumeInport.getData()->getRepresentation<VolumeRAM>();
+```
+However, you cannot directly use this representation to modify your data. This representation basically holds a `void*` pointer to the raw data. In order to safely cast it to a concrete type (`int`, `float`, `vec3`), so that you can use the data, you have to use `dispatch<>()`.
+
+The `dispatch<>()` method takes a lambda as parameter and the lambda is passed the correctly typed data as parameter. That means you can only access your data directly inside `dispatch<>()`. Furthermore, `dispatch<ReturnType>()` has a type parameter as well, which is the return type of the lambda. See the following example:
+```
+volume_ram_repr->dispatch<  std::shared_ptr<Volume>  >([] (auto ram) {
+    using ValueType = util::PrecisionValueType<decltype(ram)>;
```

Comment:
I would add a comment here saying that ValueType is the concrete type of the input, e.g. int, float...